modules/engines/engine-core/src/test/java/com/sap/xsk/modificators/XSKProjectFilesModificatorTest.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2021 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.xsk.modificators;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.io.IOUtils;
+import org.eclipse.dirigible.api.v3.platform.WorkspaceFacade;
+import org.eclipse.dirigible.core.test.AbstractDirigibleTest;
+import org.eclipse.dirigible.core.workspace.api.IFile;
+import org.eclipse.dirigible.core.workspace.api.IProject;
+import org.eclipse.dirigible.core.workspace.api.IWorkspace;
+import org.junit.Before;
+import org.junit.Test;
+
+public class XSKProjectFilesModificatorTest extends AbstractDirigibleTest {
+
+  private static IWorkspace workspace;
+  private static IProject project;
+
+  private final XSKProjectFilesModificator projectFilesModificator = new XSKProjectFilesModificator();
+
+  @Before
+  public void setUp() {
+    workspace = WorkspaceFacade.createWorkspace("test");
+    project = workspace.createProject("test");
+  }
+
+  @Test
+  public void modifyProjectFilesSuccessfully() throws IOException {
```

Comment:
too complex test, separate in few smaller test - testModifyXSJS_Successfully, testModifyXSJSLIB_Successfully, logic of reading the test file could be extracted in separate method and reused, the switch will be removed.